includes/Data/Patterns.php

```diff
@@ -304,10 +282,57 @@ public static function get_theme_step_patterns_from_step( $step, $squash = false
 				}
 			}
 		}
-
 		return $squash ? $block_patterns_squashed : $block_patterns;
 	}
 
+	/**
+	 * Retrieve Homepage Menu Step Patterns
+	 *
+	 * @return array
+	 */
+	public static function get_patterns_for_homepage_menu_slugs() {
+		$flow_data        = \get_option( Options::get_option_name( 'flow' ) );
+		$header_menu_slug = explode( '/', $flow_data['data']['partHeader'] )[1];
+		$header_slug      = ! empty( $header_menu_slug ) ? $header_menu_slug : 'site-header-left-logo-navigation-inline';
+
+		$homepage_slugs = array(
+			$header_slug,
+			'site-footer',
+		);
+
+		$homepage_style_slugs = array(
+			'homepage-1' => array(
+				'active'  => true,
+				'shown'   => true,
+				'content' => '',
+			),
+			'homepage-2' => array(
+				'active'  => true,
+				'shown'   => true,
+				'content' => '',
+			),
+			'homepage-3' => array(
+				'active'  => true,
+				'shown'   => true,
+				'content' => '',
+			),
+		);
```

Comment:
Can these be kept in the get_theme_step_patterns() function and referenced from there? cause now the slugs are not at one place but spread across functions.